mpqp/core/circuit.py

```diff
@@ -133,6 +132,11 @@ def __init__(
         self._nb_qubits: int
         """Number of qubits of the circuit."""
 
+        self.gphase: Optional[float] = None
```

Comment:
wouldn't it be simpler if it was 1 by default ? and not optional ?

Reply:
Yes we can indeed, then we check if it is different from 1, we multiply the state_vector by it at the result time, if 1 we do nothing.

Reply:
Done